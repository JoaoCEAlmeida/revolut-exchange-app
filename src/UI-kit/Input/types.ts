export interface ComponentProps {
  error?: string;
  placeholder: string;
}
