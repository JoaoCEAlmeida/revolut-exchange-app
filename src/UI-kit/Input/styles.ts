import styled from 'styled-components';

export const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const MyInput = styled.input`
  padding: 0;
  text-align: right;
  background: none;
  border-width: 0 0 0;
  outline: 0;
  border-color: ${(props) => props.theme.colors.primary.main};
  &:focus {
    outline: none;
    caret-color: ${(props) => props.theme.colors.primary.main};
  }
  ::-webkit-outer-spin-button,
  ::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
`;

export const ErrorMessage = styled.span`
  color: ${(props) => props.theme.colors.error.main};
  text-align: right;
`;
