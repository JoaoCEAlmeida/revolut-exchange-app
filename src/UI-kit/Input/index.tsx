import React from 'react';
import { MyInput, InputContainer, ErrorMessage } from './styles';
import { ComponentProps } from './types';

const Input: React.FC<ComponentProps> = ({ error, ...rest }) => {
  return (
    <InputContainer>
      <MyInput {...rest}></MyInput>
      {error && <ErrorMessage> {error} </ErrorMessage>}
    </InputContainer>
  );
};

export default Input;
