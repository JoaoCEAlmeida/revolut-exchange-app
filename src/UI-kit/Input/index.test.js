import { screen, fireEvent } from '@testing-library/react';
import Input from './index';
import renderWithProviders from '../../test/providers';

const defaultProps = {
  placeholder: 'Placeholder',
  type: 'button',
  error: ''
};

describe('Input Tests', () => {
  test('should render correctly', () => {
    const { InputTest } = renderWithProviders(() => <Input {...defaultProps} />);

    expect(InputTest).toMatchSnapshot();
  });

  test('should allow editing', () => {
    renderWithProviders(() => <Input {...defaultProps} />);
    const input = screen.getByPlaceholderText('Placeholder');

    expect(input.value).toBe('');

    fireEvent.change(input, { target: { value: '1000' } });

    expect(input.value).toBe('1000');
  });
});
