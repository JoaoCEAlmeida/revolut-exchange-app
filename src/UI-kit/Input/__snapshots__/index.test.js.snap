// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Tests should render correctly 1`] = `undefined`;
